tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -38,27 +40,87 @@ volatile uint32_t *RtosTable =
 
 void (*rtos_context_switch_ptr)();
 
+FORCE_INLINE
+void reset_erisc_info() { erisc_info->user_buffer_bytes_sent = 0; }
+
+FORCE_INLINE
+void disable_erisc_app() { flag_disable[0] = 0; }
+
+namespace internal_ {
 void __attribute__((section("code_l1"))) risc_context_switch() {
     ncrisc_noc_full_sync();
     rtos_context_switch_ptr();
     ncrisc_noc_counters_init();
 }
 
 FORCE_INLINE
-void reset_erisc_info() {
-    erisc_info->bytes_sent = 0;
+void send_fd_packets() {
+    eth_send_packet(
+        0,
+        (eth_l1_mem::address_map::ERISC_APP_RESERVED_BASE) >> 4,
+        ((eth_l1_mem::address_map::ERISC_APP_RESERVED_BASE)) >> 4,
+        (eth_l1_mem::address_map::ERISC_APP_RESERVED_SIZE) >> 4);
+    erisc_info->fd_buffer_msgs_sent += 1;
+    eth_send_packet(
+        0,
+        ((uint32_t)(&(erisc_info->fd_buffer_msgs_sent))) >> 4,
+        ((uint32_t)(&(erisc_info->fd_buffer_msgs_sent))) >> 4,
+        1);
+    while (erisc_info->fd_buffer_msgs_sent != 0) {
+        // TODO: add timer to restrict this
+        internal_::risc_context_switch();
+    }
 }
 
 FORCE_INLINE
-void disable_erisc_app() { flag_disable[0] = 0; }
+void receive_fd_packets() {
+    while (erisc_info->fd_buffer_msgs_sent != 1) {
+        // TODO: add timer to restrict this
+        internal_::risc_context_switch();
+    }
+    erisc_info->fd_buffer_msgs_sent = 0;
+    eth_send_packet(
+        0,
+        ((uint32_t)(&(erisc_info->fd_buffer_msgs_sent))) >> 4,
+        ((uint32_t)(&(erisc_info->fd_buffer_msgs_sent))) >> 4,
+        1);
+}
+
+void notify_dispatch_core_done(uint64_t dispatch_addr) {
+    for (uint32_t n = 0; n < NUM_NOCS; n++) {
+        while (!noc_cmd_buf_ready(n, NCRISC_AT_CMD_BUF))
+            ;
+    }
```

Comment:
flush both nocs because ethernet kernels could be using different nocs to try to atomically increment semaphore in dispatch core